### PR TITLE
Add src/common to include directories

### PR DESCRIPTION
**Pull request by hollasch**
This adds the src/common directory to the include directories, so
compilers will search there for headers in addition to the target source
directories.

### CODE REVIEW

**Comment by trevordblack** [comment]
This is in lieu of #347  ?

**Comment by hollasch** [comment]
Inspired by it, I suppose. It's really fixing a weakness in the original `CMakeLists.txt` that based includes from the `src/` directory, along with the default path of finding headers in the same directory as the source file. This change basically says that in addition to the source directory, also look in `src/common` for all targets. Which means that you no longer have to explicitly include something from `common/`.

**Comment by dafhi** [comment]
git compare cracks me.  i'll download locally and check it out

File: books/RayTracingInOneWeekend.html [hunk 1]
```diff
@@ -319,7 +319,7 @@
 Now we can change our main to use this:
 
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
-    #include "common/vec3.h"
+    #include "vec3.h"
```
Comment (dafhi) [inline books/RayTracingInOneWeekend.html]: overall happy with this change, though i fully support a common folder structure, which could be illustrated by a more experienced programmer than myself.

File: books/RayTracingInOneWeekend.html [hunk 2]
```diff
@@ -1064,19 +1064,20 @@
...
 
-    #include "common/ray.h"
-    #include "common/vec3.h"
+    #include "ray.h"
+    #include "vec3.h"
 
     #endif
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    [Listing [rtweekend-initial]: <kbd>[common/rtweekend.h]</kbd> The rtweekend.h common header]
+    [Listing [rtweekend-initial]: <kbd>[rtweekend.h]</kbd> The rtweekend.h common header]
```
Comment (dafhi) [inline books/RayTracingInOneWeekend.html]: rtweekend seems like a solid candidate for leaving in common

Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: No headers move in this change. It simply eliminates the need to prefix headers with the 'common/' path. It also allows programmers to put the headers all in one source directory, or to use the source code as-is with some headers in common. Either way, the compiler will locate the header.

**Pull request merged**